File: src/python/day2/input.py
import os

def readInput(text):
    result = []
    for line in text:
        if len(line) == 0:
            continue
        lineVals = []
        for v in line.split():
            lineVals.append(int(v))
        result.append(lineVals)
    return result

def readFile(file):
    with open(file) as f:
        return readInput(f.readlines())

input = readFile(os.path.join(os.path.dirname(__file__), "input.txt"))

if __name__ == "__main__":
    print(input)

File: src/rust/day8/src/input.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;
use std::collections::HashMap;

pub fn read_input(file: &str) -> (HashMap<char, Vec<(u64, u64)>>, (u64, u64)) {
    let mut result: HashMap<char, Vec<(u64, u64)>> = HashMap::new();
    let mut width: u64 = 0;
    let mut y: u64 = 0;
    if let Ok(lines) = read_lines(file) {
        for line in lines.flatten() {
            width = line.len().try_into().unwrap();
            let mut x: u64 = 0;
            for c in line.chars() {
                if c != '.' {
                    if result.contains_key(&c) {
                        result.get_mut(&c).unwrap().push((x, y));
                    } else {
                        let points = vec![(x, y)];
                        result.insert(c, points);
                    }
                }
                x += 1;
            }
            y += 1
        }
    }
    return (result, (width, y))
}

fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}
